src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs

```diff
@@ -149,36 +149,59 @@ private async Task<IPlugin> CreatePluginAsync(
             IEnumerable<string> arguments,
             IRequestHandlers requestHandlers,
             ConnectionOptions options,
+            bool isRunnablePluginFile,
             CancellationToken sessionCancellationToken)
         {
             var args = string.Join(" ", arguments);
-#if IS_DESKTOP
-            var startInfo = new ProcessStartInfo(filePath)
+
+            ProcessStartInfo startInfo;
+
+            if (!isRunnablePluginFile)
             {
-                Arguments = args,
-                UseShellExecute = false,
-                RedirectStandardError = false,
-                RedirectStandardInput = true,
-                RedirectStandardOutput = true,
-                StandardOutputEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false),
-                WindowStyle = ProcessWindowStyle.Hidden,
-            };
+#if IS_DESKTOP
+                startInfo = new ProcessStartInfo(filePath)
```

Comment:
This `ProcessStartInfo` is identical to the one when `isRunnablePluginFile` is `true`.

Therefore, if `PluginDiscover` always sets the value to `true` on .NET Framework, then this `PluginFactory` doesn't need the conditional `#if` compilation.